src/sidebar/components/group-list-item.js

```diff
@@ -79,23 +76,21 @@ function GroupListItem({
     group.type === 'private' ? 'Copy invite link' : 'Copy activity link';
 
   // Close the submenu when any clicks happen which close the top-level menu.
-  const collapseSubmenu = () => setExpanded(false);
+  const collapseSubmenu = () => onExpand(false);
 
   return (
-    <Fragment>
-      <MenuItem
-        icon={group.logo || 'blank'}
-        iconAlt={orgName(group)}
-        isDisabled={!isSelectable}
-        isExpanded={isExpanded}
-        isSelected={isSelected}
-        isSubmenuVisible={isExpanded}
-        label={group.name}
-        onClick={isSelectable ? focusGroup : toggleSubmenu}
-        onToggleSubmenu={toggleSubmenu}
-      />
-      {isExpanded && (
-        <div className="group-list-item__submenu">
+    <MenuItem
+      icon={group.logo || 'blank'}
+      iconAlt={orgName(group)}
+      isDisabled={!isSelectable}
+      isExpanded={hasActionMenu ? isExpanded : undefined}
+      isSelected={isSelected}
+      isSubmenuVisible={isExpanded}
+      label={group.name}
+      onClick={isSelectable ? focusGroup : toggleSubmenu}
+      onToggleSubmenu={toggleSubmenu}
+      renderSubmenu={() => (
```

Comment:
I've simplified this now to just having a `submenu` prop rather than `renderSubmenu`, but the original rationale was so that `MenuItem` could render the submenu only when it was wholly or partially visible.